example_allocators/bump_allocator/bump_alloc.c

```diff
@@ -51,16 +51,23 @@ void init_alloc(int size_in_bytes)
 char *bump_alloc(int len)
 {
 	char *chunk = 0;
+	size_t rounded_len; /* for CHERI alignment */
+
 	if (count + len <= max)
 	{
 		/* bump the pointer */
 		chunk = buffer + count;
 
+		/* check we can represent the capability accurately,
+		 * see p30 of CHERI C/C++ Prog Guide */
+		chunk = __builtin_align_up(chunk, ~cheri_representable_alignment_mask(len) + 1);
+		rounded_len = cheri_representable_length(len);
+
 		/* restrict capability range before returning ptr */
-		chunk = cheri_bounds_set_exact(chunk, len);
+		chunk = cheri_bounds_set_exact(chunk, rounded_len);
 
 		/* update bytes allocated count */
-		count += len;
+		count += rounded_len;
```

Comment:
Oh, i nearly missed it: I think we also need to bump by the amount that we aligned up.

`count += (chunk' - chunk) + rounded_len`

Reply:
done - thanks for feedback...